### PR TITLE
Use devkitA64 portlibs zlib library instead of builtin

### PR DESCRIPTION
**Pull request by lifajucejo**
devkitA64 provides a working zlib implementation

### CODE REVIEW

File: Makefile.libnx [hunk 1]
```diff
@@ -94,7 +93,9 @@ NO_ICON := 1
 ARCH	:=	-march=armv8-a -mtune=cortex-a57 -mtp=soft -fPIE -mcpu=cortex-a57+crc+fp+simd
 
 CFLAGS	:=	-g -Wall -O3 -ffast-math -ffunction-sections \
-			$(ARCH) $(DEFINES) -Ideps -Ideps/libz -Ilibretro-common/include -Ideps/stb -I$(LIBNX)/include -include $(LIBNX)/include/switch.h #$(shell $(DEVKITPRO)/portlibs/switch/bin/freetype-config --cflags)
+			$(ARCH) $(DEFINES) -Ideps -Ilibretro-common/include -Ideps/stb -I$(LIBNX)/include -include $(LIBNX)/include/switch.h #$(shell $(DEVKITPRO)/portlibs/switch/bin/freetype-config --cflags)
+
+INCLUDE += -I$(DEVKITPRO)/portlibs/switch/include
```
Comment (m4xw) [inline Makefile.libnx]: This should use `-I$(PORTLIBS)/include` and might be better suited for CFLAGS?

**Comment by m4xw** [comment]
@twinaphex said zlib is used for zip extraction, I know that works on some of our rnx cores.
What was your test-case?

**Comment by lifajucejo** [comment]
Tested against mame2003-plus.  Updated to use CFLAGS and env var.

**Comment by twinaphex** [comment]
Does the builtin zlib in RetroArch not work?

**Pull request closed without merging**